### PR TITLE
switch logs, idle commands to externals

### PR DESCRIPTION
**Pull request by juanvallejo**
cc @deads2k 

Depends on https://github.com/openshift/origin/pull/20362

Switches more commands to deal with external versions of objects
Picks https://github.com/kubernetes/kubernetes/pull/66352

### CODE REVIEW

File: pkg/oc/cli/logs/logs.go [hunk 1]
```diff
@@ -171,8 +209,9 @@ func (o *LogsOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []st
 		}
 		o.Options = bopts
 
-	case appsapi.Resource("deploymentconfig"),
-		appsapi.Resource("deploymentconfigs"):
+	case appsapiv1.Resource("deploymentconfig"),
+		appsapiv1.Resource("deploymentconfigs"):
+		// TODO: switch to using appsapiv1.DeploymentLogOptions once originpolymorphichelpers.LogsForObjectFn supports appsv1
```
Comment (juanvallejo) [inline pkg/oc/cli/logs/logs.go]: @deads2k fyi. Using the external api here to create the DeploymentLogOptions object would also involve changes to the logsforobject Origin polymorphic helper: https://github.com/openshift/origin/blob/master/pkg/oc/originpolymorphichelpers/logsforobject.go#L33

Comment (deads2k) [reply pkg/oc/cli/logs/logs.go]: > @deads2k fyi. Using the external api here to create the DeploymentLogOptions object would also involve changes to the logsforobject Origin polymorphic helper:

update that in this pull..

File: pkg/oc/originpolymorphichelpers/logsforobject.go [hunk 2]
```diff
@@ -39,6 +42,30 @@ func NewLogsForObjectFn(delegate polymorphichelpers.LogsForObjectFunc) polymorph
...
+		case *appsapiv1.DeploymentConfig:
+			dopts, ok := options.(*appsapiv1.DeploymentLogOptions)
+			if !ok {
+				return nil, errors.New("provided options object is not a DeploymentLogOptions")
+			}
+			appsClient, err := appsclientinternal.NewForConfig(clientConfig)
+			if err != nil {
+				return nil, err
+			}
+			return appsmanualclient.NewRolloutLogClient(appsClient.Apps().RESTClient(), t.Namespace).Logs(t.Name, *dopts), nil
```
Comment (juanvallejo) [inline pkg/oc/originpolymorphichelpers/logsforobject.go]: @deads2k not sure how to handle this - do we have a rollout client (or buildlogClient for the case below) whose Logs method receives versioned options?

Comment (mfojtik) [reply pkg/oc/originpolymorphichelpers/logsforobject.go]: i made an external version for this here https://github.com/openshift/origin/pull/20362

**Comment by DirectXMan12** [comment]
unidling changes look fine.

File: pkg/oc/cli/logs/logs_test.go [hunk 3]
```diff
@@ -12,9 +12,10 @@ import (
 	kcmd "k8s.io/kubernetes/pkg/kubectl/cmd"
 	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
 
+	buildapi "github.com/openshift/api/build/v1"
```
Comment (soltysh) [inline pkg/oc/cli/logs/logs_test.go]: `buildv1` always

File: pkg/oc/cli/logs/logs_test.go [hunk 4]
```diff
@@ -12,9 +12,10 @@ import (
...
+	buildfake "github.com/openshift/client-go/build/clientset/versioned/fake"
 	appsapi "github.com/openshift/origin/pkg/apps/apis/apps"
-	buildapi "github.com/openshift/origin/pkg/build/apis/build"
-	buildfake "github.com/openshift/origin/pkg/build/generated/internalclientset/fake"
+	internalbuildapi "github.com/openshift/origin/pkg/build/apis/build"
```
Comment (soltysh) [inline pkg/oc/cli/logs/logs_test.go]: `buildapi` always

File: pkg/oc/cli/logs/logs_test.go [hunk 5]
```diff
@@ -91,12 +92,13 @@ func TestRunLogForPipelineStrategy(t *testing.T) {
...
 			IOStreams: streams,
 			KubeLogOptions: &kcmd.LogsOptions{
+				IOStreams: streams,
 				Object:    tc.o,
 				Namespace: "foo",
 			},
 			Client: fakebc.Build(),
 		}
-		if err := o.RunLog(); err != nil {
+		if err := o.runLogPipeline(); err != nil {
```
Comment (soltysh) [inline pkg/oc/cli/logs/logs_test.go]: Why this change?

File: pkg/oc/cli/logs/logs.go [hunk 6]
```diff
@@ -13,13 +13,12 @@ import (
 	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
 	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource"
 
-	"github.com/openshift/api/apps"
-	"github.com/openshift/api/build"
-	appsapi "github.com/openshift/origin/pkg/apps/apis/apps"
+	appsapiv1 "github.com/openshift/api/apps/v1"
```
Comment (soltysh) [inline pkg/oc/cli/logs/logs.go]: `appsv1`

File: pkg/oc/cli/logs/logs.go [hunk 7]
```diff
@@ -13,13 +13,12 @@ import (
...
+	buildapiv1 "github.com/openshift/api/build/v1"
+	buildclientv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
```
Comment (soltysh) [inline pkg/oc/cli/logs/logs.go]: `buildv1client`

File: pkg/cmd/openshift-controller-manager/controller/unidling.go [hunk 8]
```diff
@@ -8,6 +8,7 @@ import (
 	appsclient "github.com/openshift/origin/pkg/apps/generated/internalclientset/typed/apps/internalversion"
 	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
 	unidlingcontroller "github.com/openshift/origin/pkg/unidling/controller"
+	kubernetes "k8s.io/client-go/kubernetes/typed/core/v1"
```
Comment (soltysh) [inline pkg/cmd/openshift-controller-manager/controller/unidling.go]: `corev1`

File: pkg/oc/cli/idle/idle.go [hunk 9]
```diff
@@ -11,18 +11,18 @@ import (
...
 	utilerrors "github.com/openshift/origin/pkg/util/errors"
+	corev1 "k8s.io/api/core/v1"
 	extensions "k8s.io/api/extensions/v1beta1"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/client-go/kubernetes"
```
Comment (soltysh) [inline pkg/oc/cli/idle/idle.go]: `clientset`

File: pkg/unidling/controller/unidling_controller.go [hunk 10]
```diff
@@ -20,6 +20,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	kubernetes "k8s.io/client-go/kubernetes/typed/core/v1"
```
Comment (soltysh) [inline pkg/unidling/controller/unidling_controller.go]: `corev1client`

File: pkg/unidling/util/scale.go [hunk 11]
```diff
@@ -3,15 +3,15 @@ package util
...
 	"github.com/golang/glog"
 
-	kapiv1 "k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
 	kextapi "k8s.io/api/extensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	kubernetes "k8s.io/client-go/kubernetes/typed/core/v1"
```
Comment (soltysh) [inline pkg/unidling/util/scale.go]: `corev1client`

File: pkg/oc/originpolymorphichelpers/logsforobject.go [hunk 12]
```diff
@@ -39,6 +45,29 @@ func NewLogsForObjectFn(delegate polymorphichelpers.LogsForObjectFunc) polymorph
...
+			}
+			if bopts.Version != nil {
+				return nil, errors.New("cannot specify a version and a build")
+			}
+			buildClient, err := buildclientv1.NewForConfig(clientConfig)
+			if err != nil {
+				return nil, err
+			}
+			return buildmanualclientv1.NewBuildLogClient(buildClient.RESTClient(), t.Namespace).Logs(t.Name, *bopts), nil
 		case *buildapi.Build:
```
Comment (soltysh) [inline pkg/oc/originpolymorphichelpers/logsforobject.go]: Additionally, since we've just migrated logs to external, do we need internals here?

File: pkg/oc/cli/logs/logs_test.go (hunk 5 above)
Comment (juanvallejo) [reply pkg/oc/cli/logs/logs_test.go]: The test `TestRunLogForPipelineStrategy` was only testing that portion of the code. Splitting it out made it easier to test
